homeassistant/components/overkiz/__init__.py

```diff
@@ -0,0 +1,120 @@
+"""The Overkiz (by Somfy) integration."""
+import logging
+
+from aiohttp import ClientError, ServerDisconnectedError
+from pyoverkiz.client import OverkizClient
+from pyoverkiz.const import SUPPORTED_SERVERS
+from pyoverkiz.exceptions import (
+    BadCredentialsException,
+    MaintenanceException,
+    TooManyRequestsException,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+from .const import (
+    CONF_HUB,
+    DOMAIN,
+    SUPPORTED_PLATFORMS,
+    UPDATE_INTERVAL,
+    UPDATE_INTERVAL_ALL_ASSUMED_STATE,
+)
+from .coordinator import OverkizDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Overkiz from a config entry."""
+    username = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+    server = SUPPORTED_SERVERS[entry.data[CONF_HUB]]
+
+    # To allow users with multiple accounts/hubs, we create a new session so they have separate cookies
+    session = async_create_clientsession(hass)
+    client = OverkizClient(
+        username=username, password=password, session=session, server=server
+    )
+
+    try:
+        await client.login()
+        setup = await client.get_setup()
+    except BadCredentialsException as exception:
+        raise ConfigEntryNotReady("Invalid authentication") from exception
+    except TooManyRequestsException as exception:
+        raise ConfigEntryNotReady("Too many requests, try again later") from exception
+    except (TimeoutError, ClientError, ServerDisconnectedError) as exception:
+        raise ConfigEntryNotReady("Failed to connect") from exception
+    except MaintenanceException as exception:
+        raise ConfigEntryNotReady("Server is down for maintenance") from exception
+    except Exception as exception:  # pylint: disable=broad-except
+        _LOGGER.exception(exception)
+        return False
+
+    coordinator = OverkizDataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="device events",
+        client=client,
+        devices=setup.devices,
+        places=setup.root_place,
+        update_interval=UPDATE_INTERVAL,
+        config_entry_id=entry.entry_id,
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    if coordinator.is_stateless:
+        _LOGGER.debug(
+            "All devices have an assumed state. Update interval has been reduced to: %s",
+            UPDATE_INTERVAL_ALL_ASSUMED_STATE,
+        )
+        coordinator.update_interval = UPDATE_INTERVAL_ALL_ASSUMED_STATE
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        "coordinator": coordinator,
+    }
```

Comment:
If the only thing being stored is the coordinator, you don't need to make a `dict`

Reply:
We store the devices per platform as well, however I left that out in this initial PR. Do I need to remove it for now? Or can I keep it as we will add this in a follow up PR.

Reply:
It is fine to leave it in since its coming later